### PR TITLE
Add extract functions

### LINKED ISSUE #106: Add additional format options for doi and pmid in enrich script
In the compose scripts from the current SYNERGY datasets, there is a lot of duplicate code for extracting the doi/pmid and adding the prefix/postfix.
I believe we can move most of these cases to the enrich script by doing the regex extraction:
doi = 10.xxxx
pmid = the numbers.
Then we can add the pre- and postfixes in the enrich scripts as well, making the process of composing the data easier.

### PR DESCRIPTION
**Pull request by EmilyWes**
Added 4 functions to easily extract/augment identifiers.
Each function does the following things:
- Takes data from an input column
- Possibly apply a regex filter or do retyping.
- Possibly adding a prefix url
- Add those values in the identifier column, optionally overwriting existing data.
Ruff checked and formatted.
Closes #106

### CODE REVIEW

File: scripts/utils.py [hunk 1]
```diff
@@ -60,6 +60,83 @@ def rename_columns(
...
+    else:
+        # Only overwrite rows that did not have a doi yet.
+        df["doi"] = df["doi"].mask(
+            df["doi"].isnull(), url_prefix + df[col_in].str.extract(regex)[0]
+        )
+
+    return df
+
+10\.[^&]*
+def extract_pmid(
```
Comment (J535D165) [inline scripts/utils.py:86]: ☝️

File: scripts/utils.py [hunk 2]
```diff
@@ -60,6 +60,83 @@ def rename_columns(
...
+
+
+def extract_year(df: pd.DataFrame, col_in: str, overwrite: bool = False):
+    """Extracts the number from the input column and stores it in year column."""
+
+    regex = r"(\d+)"
+
+    if ("year" not in list(df)) or overwrite:
+        df["year"] = df[col_in].str.extract(regex)
+    elif df[col_in].dtype == "object":
```
Comment (J535D165) [inline scripts/utils.py:116]: Better use this: `from pandas.api.types import is_string_dtype`. pandas is changing a lot for v3 on string columns

**Comment by J535D165** [comment]
Thanks looks good to me

File: scripts/utils.py (hunk 2 above)
Comment (EmilyWes) [reply scripts/utils.py]: Nice that they are working on strings

**Comment by EmilyWes** [comment]
Thanks for the quick review! PR is updated.

**Review by J535D165 — APPROVED** [review]

**Comment by J535D165** [comment]
Thanks!

**Pull request merged**